### DISCUSSION TITLE
Repository And Endpoint Custom Service

### DISCUSSION THREAD

**Discussion by mgdm187** [serenity-is/Serenity#5931] [Help]
Hello,
I am new to Serenity and I always ask when I'm not sure or when I don't know something. I would like to know how to create new custom service on XYZRepository or XYZEndpoint. I'm confused which is for which and if you have some example on how to create custom service that I can call to update multiple rows in tables at once.
Every clarification on how repository and custom service work would be great.
Thank you in advance.
BR

**Comment by edwardch** [comment]
Your grid or dialog will call the service that's in the Endpoint, then that will call your method in your Repository.

[https://github.com/serenity-is/common-features/tree/master/src/Serenity.Demo.Northwind/Serenity.Demo.Northwind/Customer](https://github.com/serenity-is/common-features/tree/master/src/Serenity.Demo.Northwind/Serenity.Demo.Northwind/Customer)

Look at the endpoint and repository files in this directory. You should customise your own code in your repo. You can also look in the the UserRepository for some inspiration.
